CMakeLists.txt

```diff
@@ -13,19 +13,55 @@ set(SO_MAJOR 2)
 set(SO_MINOR 0)
 set(SO_PATCH 2)
 
-add_definitions (
-  -DUTF8PROC_EXPORTS
-)
+# configure build type
+
+set(_utf8proc_is_set_build_shared_libs TRUE)
+if(DEFINED BUILD_SHARED_LIBS)
+  set(_utf8proc_is_set_build_shared_libs ${BUILD_SHARED_LIBS})
+endif(DEFINED BUILD_SHARED_LIBS)
+
+set(UTF8PROC_BUILD_AS_SHARED ${_utf8proc_is_set_build_shared_libs} CACHE BOOL "If present and true, this will cause utf8proc to be built shared unless the utf8proc was explicitly added as a static library.")
+
+# set additional defined
 
-if (NOT MSVC)
+if(NOT DEFINED _utf8proc_additional_defined)
+  set(_utf8proc_additional_defined )
+endif(NOT DEFINED _utf8proc_additional_defined)
+
+if(UTF8PROC_BUILD_AS_SHARED)
+  list(APPEND _utf8proc_additional_defined 
+    "-DUTF8PROC_EXPORTS"
+  )
+  set(_utf8proc_build_type "SHARED")
+else(UTF8PROC_BUILD_AS_SHARED)
+  list(APPEND _utf8proc_additional_defined 
+    "-DUTF8PROC_BUILD_AS_LIB"
```

Comment:
`BUILD_AS_STATIC` seems more appropriate as the platform-independent antonym of "shared"